### DISCUSSION TITLE
PN532 & HomeSpan repos

### DISCUSSION THREAD

**Discussion by jklap** [rednblkx/HomeKey-ESP32#28] [General]
@rednblkx looking at your changes in these two repos, are you planning on trying to get them merged upstream?

It would seem the HomeSpan changes would likely be accepted?

Lots of renaming in PN532 which makes it hard pull in the changes from upstream -- I'm wondering if it might be worth reverting to the original directory structure and then trying to get your changes merged into upstream? They seem like they are clean enough and useful enough to be accepted?

**Comment by rednblkx** [comment]
I do plan to merge my changes upstream, however, first i want to replace some "quick hacks" that i did with proper code before making a PR but just never got around to do it.

About the PN532 repo, i've had some issues with the library and PIO so i just moved everything into one folder as i was sick of it 😮‍💨 and i know there was something with the HSU module, don't remember what exactly so i just removed it and put it on the pile of "things i'll deal with later 🙃" like everything above. Also, i've looked on the upstream repo, their last commit was 2 years ago and they have multiple open PRs from 2022 with zero comments, doubt they are still watching the repo.

If you wish to contribute, you can open a PR and i'll take a look when i can.
